### ISSUE TITLE
Some Z-layers are never rendered in FPS mode.

### ISSUE THREAD

**Issue by alexchandel** [JapaMala/armok-vision#59]
When switching to FPS mode on a certain layer, then moving up or down a layer, some Z-layers are never rendered. Notice here, multiple tree layers aren't shown:

<img width="1680" alt="screen shot 2017-11-04 at 2 00 15 pm" src="https://user-images.githubusercontent.com/637714/32408627-b8fca9a2-c171-11e7-9e40-0c768b3c31fa.png">

**Comment by JapaMala** [comment]
There are a number of possible causes. Most likely is just the fact that updating the meshes is currently a lot slower than it could be, in which case waiting for those layers to get generated should work.

Otherwise if there's an error in the meshing, I would need to look into that.

**Comment by alexchandel** [comment]
It's always a few Z layers above (and maybe below) where you spawn in FPS mode. And layers above the missing Z layers always spawn (notice the tops of the trees in that pictures).

Do you regenerate the entire mesh whenever the god-mode camera is moved up or down, or just hide certain layers? Is the mesh deleted & regenerated when the camera moves around in FPS mode?

**Comment by JapaMala** [comment]
Is this problem still there in 0.18.0?

I made changes to the visibility there, which should have fixed this.

**Comment by alexchandel** [comment]
Yes, it just happened to me about a week ago in 0.18.0. I'm waiting to test a build that includes the new update timers, to see if it's due to extremely slow mesh updating.

**Comment by alexchandel** [comment]
I may have found the problem. Does AV *always* render based on where the FPS camera is in FPS mode?

I can't always replicate it, but after cutting my "Draw Range Side" down to 2, I'm sometimes able to reach the edge. Upon returning to god mode, the god mode camera hadn't moved. I don't think it's due to mesh generation lag either, because AV never rose above 35% CPU during that time (though inputs still lag).

**Comment by alexchandel** [comment]
After more testing, this seems to happen after the god mode camera has moved, when following the DF screen.
